src/content/reference/device-os/typed-publish.md

```diff
@@ -0,0 +1,177 @@
+
+```
+data:application/octet-stream;base64,pyKYHEAbm4C7ndnAE7tO0KPAroHFk5Eqg45pJ7DGFyaFk7em9WnATJ49U0m1R/BEJpuKHeS8c/lNpOg0wlYXyQ==
+```
+
+This is encoded in the [Data URL](https://developer.mozilla.org/en-US/docs/Web/URI/Schemes/data) format. 
+
+Specifically, the cloud encodes binary data in Base 64 in a Data URL. This can easily be parsed by most services. Since the Base 64 data is still
+text and valid JSON, it is compatible with both the default webhook template, as well as custom templates that use Mustache templates.
+
+```json
+{
+  "event": "\{{{PARTICLE_EVENT_NAME}}}",
+  "data": "\{{{PARTICLE_EVENT_VALUE}}}",
+  "coreid": "\{{{PARTICLE_DEVICE_ID}}}",
+  "published_at": "\{{{PARTICLE_PUBLISHED_AT}}}"
+}
+```
+
+## Structured data
+
+Particle has long-recommended using JSON format for sending data from your device to the cloud using Particle.publish.
+
+With Device OS 6.2 and later, you can now send structured data more efficiently over-the-air without affecting your
+back-end services and webhooks.
+
+Say, for example, your device publishes this JSON data currently:
+
+```json
+{"a":123,"b":"test","c":true,"d":[1,2,3]}
+```
+
+This is 41 characters of JSON data in the compact form, above.
+
+With Device OS 6.2 structured data, this will be converted into CBOR format instead of JSON over-the-air. 
+[CBOR](https://cbor.io/), the Concise Binary Object Representation (RFC 8949) can be converted to and from
+JSON without loss.
+
+The JSON data above encodes to 21 bytes of binary CBOR data, a savings of 51%!
+
+Since CBOR and be converted to and from JSON transparently, that is exactly what the cloud does before sending the data to
+webhooks and SSE. Your webhook will continue to see JSON data as before, but it will be transmitted more efficiently over-the-air.
+
+Furthermore, the 1024 byte publish limit in 6.2.x and earlier occurs after the conversion to CBOR, so you can publish significantly
+larger JSON data than before.
+
+To see how much smaller your JSON data will be in CBOR, you can use the [JSON tool](/tools/developer-tools/json/) which can now convert to CBOR and compare
+the sizes.
+
+If you have a JSON object and want to create code stubs for generating a `Variant` with the same
+shape, the [JSON tool](/tools/developer-tools/json/) also has a Variant code generator.
+
+
+### Structured binary data
+
+Additionally, you can include binary data ("buffer") in your structured data. This is kept in binary format in the CBOR data stream so it
+does not use additional data over-the-air, as was the case when encoding to hex, Base64, or Base85 on-device. When the cloud 
+receives the binary data, it expands it into a JSON object containing meta information and Base64 encoded data. This allows
+for easy processing in Logic, webhooks, or SSE.
+
+For example, you could use this code:
+
+```cpp
+VariantMap map;
+map.set("a", Variant(1234));
+map.set("b", Variant(Buffer::fromHex("9dcae4dfe9af57338b5755d67e51771c")));
```

Comment:
This example can be simplified:

```cpp
Variant map;
map.set("a", 1234);
map.set("b", Buffer::fromHex("9dcae4dfe9af57338b5755d67e51771c"));
```
For convenience, `Variant` exposes some of the methods of `VariantMap` and `VariantArray` so that it's not necessary to cast to a different type when working with maps and arrays. Also, all constructors of `Variant` are implicit so there's usually no need to them explicitly.

Reply:
Oh, that's really useful! Fixed.